src/app.controller.ts

```diff
@@ -0,0 +1,88 @@
+import {
+  Controller,
+  Get,
+  Param,
+  Post,
+  Res,
+  StreamableFile,
+  UploadedFile,
+  UseInterceptors,
+} from '@nestjs/common';
+import { FileInterceptor } from '@nestjs/platform-express';
+import { AppService } from './app.service';
+import { storage } from './config/storage.config';
+import * as fs from 'fs';
+import { CSVtoPDF } from './tryTable';
+
+@Controller()
+export class AppController {
+  constructor(private readonly appService: AppService) {}
+  private outpath: string;
+  @Get()
+  sayHello() {
+    return this.appService.getHello();
+  }
+
+  @Post('upload')
+  @UseInterceptors(FileInterceptor('file', { storage }))
+  async upload(@UploadedFile() file) {
+    const createPDF = new CSVtoPDF();
+    const stream = fs.createWriteStream('./document.pdf');
+    this.outpath = createPDF.conversionTop(
+      file.destination + '/' + file.filename,
+      stream,
+    );
+    return {
+      fileName: file.originalname,
+      // savedPath: path.replace(/\\/gi, '/'),
+      size: file.size,
+    };
```

Comment:
This is not the correct method. You should set Response header as `Content-disposition` as `attachment; filename=<your filename>` and `Content-Type` as `application/pdf`.

Reply:
This guide will also help: https://docs.nestjs.com/techniques/streaming-files

Reply:
This is a hint:

```typescript
// set download filename
            const filename = // your filename;
            response.setHeader('Content-disposition', `attachment; filename=${filename}.pdf`);
            response.set('Content-Type', 'application/pdf');
            response.setTimeout(1800 * 1000); // 30 minutes timeout

            // make stream pipe
            const input = new Readable({
                objectMode: true,
            });
            input._read = () => { };

            const columnLabels = [
                "id",
                "market_id"
            ];

            const COLUMNS = [];
            columnLabels.forEach(label => COLUMNS.push({label, value: label}));

            const opts = {fields: COLUMNS};
            const transformOpts = {objectMode: true};
            const json2csv = new Transform(opts, transformOpts);

            input.pipe(json2csv).pipe(response);
```